Package.swift

```diff
@@ -1,13 +1,4 @@
-// swift-tools-version:5.1
-
-///  To avoid breaking Swift 5.1 compatibility, only require newer NIO versions when we need NIO's concurrency APIs.
-func getMinNIOVersion() -> PackageDescription.Package.Dependency.Requirement {
-#if compiler(>=5.5.2) && canImport(_Concurrency)
-    return .upToNextMajor(from: "2.36.0")
-#else
-    return .upToNextMajor(from: "2.15.0")
```

Comment:
I could have left this in and updated this to be NIO 2.32 (the minimum version where the separate modules exist), but there didn't seem to be a huge benefit to doing so as 2.36.0 supports 5.2+, so I thought it was simpler and reasonable to remove the extra complexity here and require everyone to be on 2.36+.